hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieIndexer.java

```diff
@@ -149,19 +149,18 @@ public static void main(String[] args) {
 
     if (cfg.help || args.length == 0) {
       cmd.usage();
-      System.exit(1);
+      throw new HoodieException("Indexing failed for basePath : " + cfg.basePath);
     }
 
     final JavaSparkContext jsc = UtilHelpers.buildSparkContext("indexing-" + cfg.tableName, cfg.sparkMaster, cfg.sparkMemory);
     HoodieIndexer indexer = new HoodieIndexer(jsc, cfg);
     int result = indexer.start(cfg.retry);
     String resultMsg = String.format("Indexing with basePath: %s, tableName: %s, runningMode: %s",
         cfg.basePath, cfg.tableName, cfg.runningMode);
-    if (result == -1) {
-      LOG.error(resultMsg + " failed");
-    } else {
-      LOG.info(resultMsg + " success");
+    if (result != 0) {
+      throw new HoodieException(resultMsg + " failed");
     }
+    LOG.info(resultMsg + " success");
     jsc.stop();
```

Comment:
Same here, I think we can add a try catch block here to make sure jsc exit gracefully

Reply:
Looks like `HoodieIndexer` also uses `UtilHelpers.retry`

Reply:
Yes. `HoodieCompactor/HoodieClusteringJob/HoodieIndexer` use `UtilHelpers.retry`, so it's no need to add try catch block in main method.